src/libcollections/vec_deque.rs

```diff
@@ -2121,6 +2122,42 @@ impl<T: fmt::Debug> fmt::Debug for VecDeque<T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.9.0")]
+impl<T> From<Vec<T>> for VecDeque<T> {
+    fn from(other: Vec<T>) -> Self {
+        unsafe {
+            let other_buf: *mut T = mem::transmute(other.as_ptr());
+            let buf = RawVec::from_raw_parts(other_buf, other.capacity());
+            let len = other.len();
+            mem::forget(other);
+
+            let mut out = VecDeque {
+                tail: 0,
+                head: len,
+                buf: buf
+            };
+
+            // Do this to force resize to correct size;
+            // otherwise we may get not_power_of_two assert failures!
+            out.reserve(0);
```

Comment:
It'd be good to test for the power-of-two invariant in your test for this method.